### PR TITLE
biomes and trash rework.

### PR DESCRIPTION
**Pull request by Evankhell561**
## About The Pull Request

the trash piles now work based on spawn tags.
add biome spawners.

## Why It's Good For The Game
Because kotmap.

## Changelog
:cl:
add: biome spawners: These are a smart spawner that prioritizes the most dangerous mobs and traps depending on the amount of loot in the area. 
tweak: several deep maint maps work with biomes.
tweak: the trash piles now work on spawn tags.
tweak: trash piles now always contain at least one rare item.
/:cl:

### CODE REVIEW

**Comment by Evankhell561** [comment]
Although this pr is based on the documentation I will leave the "oldification code rework" part for a separate pr.

**Review by genessee596 — APPROVED** [review]
Delegate stamp of approval.

Yes

Still needs maintainer approval.

**Comment by genessee596** [comment]
Make your description more.. well, descriptive, please.

File: code/game/objects/random/biomes/biome.dm
Comment (TheShown911) [inline code/game/objects/random/biomes/biome.dm:11]: cap_price / 10 would equal 50000

**Review by TheShown911 — COMMENTED** [review]
Testing required.

File: code/game/objects/random/biomes/biome.dm
Comment (Evankhell561) [reply code/game/objects/random/biomes/biome.dm:11]: this variable is automatically set to the variable "cap price / 10" when the object is created.

**Comment by TheShown911** [comment]
Oddities in junk piles seam way too common.

**Comment by genessee596** [comment]
Oddities are incredibly common. Please make them appear less.

**Review by TorinoFermic — APPROVED** [review]

**Comment by TorinoFermic** [comment]
Code was tested last monday, merging.

**Pull request merged**
